nsa/demo.py

```
@@ -0,0 +1,230 @@
+
+    with torch.no_grad():
+        output = model(img_tensor)
+        pred = output.argmax(dim=1).squeeze().cpu().numpy()
+
+    return pred
+
+
+def visualize_results(
+    original: np.ndarray,
+    prediction: np.ndarray,
+    save_path: str = None,
```

Ruff (0.14.8)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)